both/gamelogic.js

```diff
@@ -137,100 +137,186 @@ GameLogic = {
+      var tile = Tiles.getBoardTile(player.position.x, player.position.y,players.length,game.name);
+      if (tile.tileType === Tiles.ROLLER && tile.speed === 2) {
+        GameLogic.movePlayerWithRoller(player, players, tile);
+      }
+    });
+    scope.tryToMovePlayersOnRollers(players, roller_moves);
+    callback();
+  }
+  
+  scope.executeGears = function(players, callback) {
+    var game = Games.findOne(players[0].gameId);
+    players.forEach(function(player) {
+      var tile = Tiles.getBoardTile(player.position.x, player.position.y,players.length,game.name);
+      if (tile.tileType === Tiles.GEAR) {
+        player.direction = tile.rotate;
+        player.direction %= 4;
+        checkRespawnsAndUpdateDb(players, player, _CARD_PLAY_DELAY);
+      }
+    });
+    callback();
+  }
+  
+  scope.executePushers = function(players, callback) {
+    var game = Games.findOne(players[0].gameId);
+    players.forEach(function(player) {
+      var tile = Tiles.getBoardTile(player.position.x, player.position.y,players.length,game.name);
+      if (tile.tileType === Tiles.PUSHER &&  game.playPhaseCount % 2 === tile.pusher_type ) {
+        tryToMovePlayer(players, player, tile.move, game);
+        checkRespawnsAndUpdateDb(players, player, _CARD_PLAY_DELAY);
+      }
+    });
+    callback();
+  }
+  
+  scope.executeLasers = function(players, callback) {
+    var game = Games.findOne(players[0].gameId);
+    players.forEach(function(player) {
+      var tile = Tiles.getBoardTile(player.position.x, player.position.y,players.length,game.name);
+      if (tile.damage > 0) {
+        player.damage += tile.damage;
+        checkRespawnsAndUpdateDb(players, player, _CARD_PLAY_DELAY);
+      }
+      if (!player.powered_down) {
+        scope.shootRobotLaser(players, player, players.length, game.name);
+      }
+    });
+    callback();
+  }
+  
+  scope.shootRobotLaser = function(players, player, playerCount, boardName ) {
+    var stepY = 0;
+    var stepX = 0;
+    var wallDir = [];
+    switch (player.direction) {
+      case GameLogic.UP:
+        stepY = -1;
+        wallDir = ['up','down'];
+        break;
+      case GameLogic.RIGHT:
+        wallDir = ['right','left'];
+        stepX = 1;
+        break;
+      case GameLogic.DOWN:
+        stepY = 1;
+        wallDir = ['down','up'];
+        break;
+      case GameLogic.LEFT:
+        wallDir = ['left','right'];
+        stepX = -1;
+        break;
+    }
+    var x = player.position.x;
+    var y = player.position.y;
+    while ( x > 0 && y > 0 && x < Tiles.BOARD_WIDTH && y < Tiles.BOARD_HEIGHT &&
+            !Tiles.hasWall(x,y, wallDir[0], playerCount, boardName) && !Tiles.hasWall(x+stepX,y+stepY, wallDir[1],playerCount, boardName)) {
+      x += stepX;
+      y += stepY;
+      victim = Tiles.isPlayerOnTile(players,x,y)
+      if (victim) {
+        victim.damage += 1;
+        checkRespawnsAndUpdateDb(players, victim, _CARD_PLAY_DELAY);
+        break;
+      }
```

Comment:
sorry, I didn't use the new lineOfSight method. Seemed more convenient to me to just follow each robots direction until a wall or another robot is hit.